src/app/(dashboard)/interview/questionnaire/page.tsx

```diff
@@ -0,0 +1,279 @@
+    options: [
+      "undefined", // Correct answer
+      "null",
+      "0",
+      "NaN",
+    ],
+  },
+  {
+    question: "Which of the following is a feature of functional programming?",
+    options: [
+      "State is mutable",
+      "Functions are first-class citizens", // Correct answer
+      "Objects are the main focus",
+      "Using loops extensively",
+    ],
+  },
+];
+
+interface QuestionnaireProps {
+  hasAnswered?: boolean;
+}
+
+const Questionnaire: React.FC<QuestionnaireProps> = ({ hasAnswered }) => {
+  if (hasAnswered) {
+    return <Interview hasAnswered />;
+  }
+
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [selectedOptions, setSelectedOptions] = useState(
+    Array(questions.length).fill(null)
+  );
+
+  const handleOptionSelect = (option: string) => {
+    const updatedSelectedOptions = [...selectedOptions];
+    updatedSelectedOptions[currentQuestionIndex] = option;
+    setSelectedOptions(updatedSelectedOptions);
+  };
+
+  const handleNext = () => {
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex(currentQuestionIndex + 1);
+    }
+  };
+
+  const handleBack = () => {
+    if (currentQuestionIndex > 0) {
+      setCurrentQuestionIndex(currentQuestionIndex - 1);
+    }
+  };
+
+  return (
+    <div className="max-w-2xl mx-auto p-6 bg-white shadow-md rounded-lg">
+      <h1 className="font-bold text-2xl mb-6">Job Interview Questionnaire</h1>
+      <h2 className="text-lg font-semibold mb-4">
+        Question {currentQuestionIndex + 1} of {questions.length}:{" "}
+        {questions[currentQuestionIndex].question}
+      </h2>
+      <div className="space-y-2">
+        {questions[currentQuestionIndex].options.map((option, index) => (
+          <button
+            key={index}
+            className={`w-full px-4 py-2 rounded-lg hover:bg-gray-300 ${
+              selectedOptions[currentQuestionIndex] === option
+                ? "bg-blue-500 text-white"
+                : "bg-gray-200"
+            }`}
+            onClick={() => handleOptionSelect(option)}
+          >
+            {option}
+          </button>
+        ))}
+      </div>
+      <div className="flex justify-between mt-6">
+        <button
+          onClick={handleBack}
+          className={`px-4 py-2 bg-gray-300 rounded-lg hover:bg-gray-400 ${
+            currentQuestionIndex === 0 ? "opacity-50 cursor-not-allowed" : ""
+          }`}
+          disabled={currentQuestionIndex === 0}
+        >
+          Back
+        </button>
+        <button
+          onClick={handleNext}
+          className={`px-4 py-2 bg-gray-300 rounded-lg hover:bg-gray-400 ${
+            currentQuestionIndex === questions.length - 1
+              ? "opacity-50 cursor-not-allowed"
+              : ""
+          }`}
+          disabled={currentQuestionIndex === questions.length - 1}
+        >
+          Next
+        </button>
```

Comment:
Need to add finish button and modals to restrict users going to other pages while quiz is happening